care/facility/api/viewsets/asset.py

```diff
@@ -412,6 +412,52 @@ def operate_assets(self, request, *args, **kwargs):
                 status=status.HTTP_500_INTERNAL_SERVER_ERROR,
             )
 
+    @extend_schema(tags=["asset"])
+    @action(methods=["POST"], detail=False)
+    def bulk_upsert(self, request, *args, **kwargs):
+        if "assets" not in request.data:
+            raise ValidationError({"assets": "No Data was provided"})
+        if not isinstance(request.data["assets"], list):
+            raise ValidationError({"assets": "Invalid Data"})
+        if "location" not in request.data:
+            raise ValidationError({"location": "Location is required"})
+
+        limit = request.data.get("limit", None)
+        if limit is None:
+            limit = float("inf")
+        if len(request.data["assets"]) > limit:
+            raise ValidationError(
+                {"assets": f"Maximum of {limit} assets can be created at once"}
+            )
```

Comment:
I don't see any use case where FE/client would need to specify a limit, in order to limit number of rows added 😅 

Limit needs to be limited in backend. Why would client specify a limit if it can anyways be infinite?

cc: @sainak

Reply:
@rithviknishad @sainak ? should i hard code the limit for now to 2000? or completely remove the limit as we have in external result bulk upsert